var/spack/repos/builtin/packages/parflow/package.py

```diff
@@ -65,21 +65,19 @@ def setup_run_environment(self, env):
 
     examples_dir = "examples"
 
-    def test(self):
+    def test_parflow(self):
         """Perform smoke test on installed ParFlow package."""
         # Run the single phase flow test
         run_path = join_path(self.spec.prefix, self.examples_dir)
         if os.path.isdir(run_path):
+            options = ["default_single.tcl", "1", "1" "1"]
             with working_dir(run_path):
-                self.run_test(
-                    "{0}/tclsh".format(self.spec["tcl"].prefix.bin),
-                    ["default_single.tcl", "1", "1" "1"],
-                )
+                exe = which("{0}/tclsh".format(self.spec["tcl"].prefix.bin))
+                exe(*options)
         else:
             # If examples are not installed test if exe executes
-            exes = ["parflow"]
+            exes = [join_path(self.prefix.bin, "parflow")]
             for exe in exes:
-                reason = "test version of {0} is {1}".format(exe, self.spec.version)
-                self.run_test(
-                    exe, ["-v"], [self.spec.version.string], installed=True, purpose=reason
-                )
+                exe = which(exe)
+                out = exe("-v", output=str.split, error=str.split)
+                assert self.spec.version in out
```

Comment:
This looks like it should be a separate test method (e.g., test_check_version).  There's also no reason to have a for loop over the `exes` since there's only one ATM.